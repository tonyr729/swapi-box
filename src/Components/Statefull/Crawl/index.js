import React, { Component } from 'react';
import APIManager from '../../../Helpers/API/API Manager/apiManager'
import './index.css'


class Crawl extends Component {
  constructor() {
    super();
    this.state = {
      episode: '',
      title: '',
      crawl: '',
      release: ''
    }
  }

  async componentDidMount(){
    const randomNumber = Math.floor(Math.random() * 7) + 1;
    const api = new APIManager();
    const crawl = await api.fetchCrawl(randomNumber)
    this.setCrawl(crawl)
  }

  setCrawl = ({episode, title, crawl, release}) => {
    this.setState({ episode, title, crawl, release})
  }


  render(){ 
    return (
      <div className="crawl-container">
        <div className="fade"></div>
        <section className="star-wars">
          <div className="crawl">
          <div className="title">
            <p>Episode {this.state.episode}</p>
            <h1>{this.state.title}</h1>
          </div>
            <p>{this.state.crawl}</p>
            <p>{this.state.release}</p>
          </div>
        </section>
        {/* <p className="crawl-skip">Press Enter to skip</p> */}
      </div>
    )
  }
}



export default Crawl;