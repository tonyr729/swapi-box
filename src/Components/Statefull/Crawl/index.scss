body {
  width: 100%;
  height: 100%;
  overflow: hidden;
  background: #000;
}

.star-wars {
  display: flex;
  justify-content: center;
  height: 800px;
  perspective: 400px;
  color: #feda4a;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 500%;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 150%;
  text-align: justify;
}

.crawl {
  position: relative;
  top: 300px;
  transform-origin: 50% 100%;
  animation: crawl 40s linear;
  visibility: hidden;
}

@keyframes crawl {
  0% {
    top: 300;
    transform: rotateX(20deg) translateZ(0);
    visibility: visible;
  }

  100% { 
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}

.fade {
  position: relative;
  width: 100%;
  min-height: 60vh;
  top: -25px;
  background-image: linear-gradient(0deg, transparent, black 75%);
  z-index: 1;
}