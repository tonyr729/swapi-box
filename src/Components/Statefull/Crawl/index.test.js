import React from 'react';
import ReactDOM from 'react-dom';
import APIManager from '../../../Helpers/API/API Manager/apiManager';
import Crawl from './index'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
jest.mock('../../../Helpers/API/API Manager/apiManager.js', () => {
  return jest.fn().mockImplementation(() => {
    return {
      fetchCrawl: jest.fn().mockImplementation(() => Promise.resolve("Kitten")),
    };
  });
});

describe('Main', () => {

  describe('Component', () => {
    it('should match snapshot', () => {
      const crawl = renderer.create(<Crawl />).toJSON();

      expect(crawl).toMatchSnapshot();
    })
  })
})
