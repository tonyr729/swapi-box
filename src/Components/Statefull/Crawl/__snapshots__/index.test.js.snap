// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Crawl Component should match snapshot 1`] = `
<div
  className="crawl-container"
>
  <div
    className="fade"
  />
  <section
    className="star-wars"
  >
    <div
      className="crawl"
    >
      <div
        className="title"
      >
        <p>
          Episode 
          
        </p>
        <h1>
          
        </h1>
      </div>
      <p>
        
      </p>
      <p>
        
      </p>
    </div>
  </section>
</div>
`;
